packages/lab/src/slider/Slider.tsx

```diff
@@ -1,127 +1,174 @@
-    controlled: parseValueProp(valueProp, min, max),
+  const [valueState, setValue] = useControlled({
+    controlled: valueProp,
     default: defaultValue,
     name: "Slider",
-    state: "Value",
+    state: "value",
   });
 
-  const handleSliderChange = (value: SliderValue) => {
-    setValue(value);
-    onChange?.(value);
+  const {
+    handlePointerDownOnThumb,
+    handlePointerDownOnTrack,
+    calculateAndSetThumbPosition,
+    calculatePercentage,
+    clamp,
+    isDragging,
+    sliderRef,
+  } = useSliderThumb({
+    min,
+    max,
+    step,
+    valueState,
+    // @ts-ignore onChange can accept both number and [number, number] value types
+    onChange,
+    // @ts-ignore setValue can accept both number and [number, number] value types
+    setValue,
+  });
+
+  const {
+    a11yProps: {
+      "aria-labelledby": formFieldLabelledBy,
+      "aria-describedby": formFieldDescribedBy,
+    } = {},
+    disabled: formFieldDisabled,
+  } = useFormFieldProps();
+
+  const disabled = formFieldDisabled || disabledProp;
+  const value = clamp(valueState);
+  const progressPercentage = calculatePercentage(toFloat(value));
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const parsedValue = toFloat(event.target.value);
+    setValue(parsedValue);
+    onChange?.(event, parsedValue);
   };
 
   return (
-    <SliderContext.Provider
-      value={{
-        value,
-        min,
-        max,
-        step,
-        onChange: handleSliderChange,
-        ariaLabel,
-      }}
+    <SliderTrack
+      disabled={disabled}
+      format={format}
+      handlePointerDown={handlePointerDownOnTrack}
+      labelPosition={labelPosition}
+      isDragging={isDragging}
+      min={min}
+      minLabel={minLabel}
+      max={max}
+      maxLabel={maxLabel}
+      markers={markers}
+      progressPercentage={progressPercentage}
+      ref={ref}
+      sliderRef={sliderRef}
+      {...rest}
     >
-      <div
-        ref={ref}
-        className={clsx(
-          withBaseName(),
-          { [withBaseName("bottomLabel")]: marks !== "inline" },
-          className,
-        )}
-        {...rest}
-      >
-        {marks !== "all" && (
-          <Label
-            className={clsx(withBaseName("label"), {
-              [withBaseName("labelMinBottom")]: marks !== "inline",
-            })}
-          >
-            {min}
-          </Label>
-        )}
-        <SliderTrack />
-        {marks !== "all" && (
-          <Label
-            className={clsx(withBaseName("label"), {
-              [withBaseName("labelMaxBottom")]: marks !== "inline",
-            })}
-          >
-            {max}
-          </Label>
-        )}
-        {marks === "all" && <SliderMarks max={max} min={min} step={step} />}
-      </div>
-    </SliderContext.Provider>
+      <SliderThumb
+        aria-describedby={formFieldDescribedBy}
+        aria-label={ariaLabel}
+        aria-labelledby={formFieldLabelledBy}
```

Comment:
aria-labelledby and aria-describedby being passed in as props should also be handled (and tested)